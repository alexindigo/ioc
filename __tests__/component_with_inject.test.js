/* eslint-env jest */

/*
 * Individual component testing is pretty simple
 * just provide your dependencies as props
 * and add `.dive()` step to your shallow render,
 * as with any High Order Component.
 *
 * Remarks about `.html()` may apply,
 * depending if any of the children components
 * expect anything from the context
 */

import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import Component from './components/Endpoint';

describe('With Enzyme', () => {
  it('Component renders with props', () => {
    // no need to mock Tracking component much for shallow rendering
    const injected = shallow(<Component AB={'B'} Tracking={() => {}} />);
    const component = injected.dive();
    expect(component.find('h3').text()).toEqual('Endpoint (B)');
    expect(component.find('Tracking').first().find('a').text()).toEqual('About: foo baz');
  });
});

describe('With Snapshot Testing', () => {
  it('Component renders with props', () => {
    const component = renderer.create(<Component AB={'B'} Tracking={(props) => <div comment="mocked Tracking component">{props.children}</div>} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
