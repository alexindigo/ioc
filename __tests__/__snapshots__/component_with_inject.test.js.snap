// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing Component renders with props 1`] = `
<div
  style={
    Object {
      "border": "1px dashed #00ff00",
      "marginTop": "5px",
      "padding": "10px",
    }
  }
>
  <h3>
    Endpoint (
    B
    )
  </h3>
  Uses injected \`Log\` component without direct dependency on one
  <br />
  <div
    comment="mocked Tracking component"
  >
    <a>
      About: foo baz
    </a>
  </div>
</div>
`;
